paperqa/agents/search.py

```diff
@@ -674,6 +699,7 @@ async def get_directory_index(  # noqa: PLR0912
     )
     with progress_bar:
         async with anyio.create_task_group() as tg:
+            processed_counter: Counter = Counter()
```

Comment:
```suggestion
            processed_counter = Counter()
```

No need to repeat the type hint, Python is not strongly typed

Reply:
I think this was because of the lint

Reply:
Ah I see. Just to be precise, it's not really "lint", it's the type checker `mypy` (which would be `pre-commit`)

I think then let's do this

```suggestion
            processed_counter: Counter[str] = Counter()
```

The reason is it's telling `mypy` what type to expect in the `Counter`: https://github.com/python/typeshed/blob/aac4394eb29d86797628b57d6f01f5e17f5ff83f/stdlib/collections/__init__.pyi#L277

Reply:
<img width="529" alt="image" src="https://github.com/user-attachments/assets/35fa8487-696d-43fd-8ffe-114591f5c9e6" />